File: programming-skills-study-plan-ii/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        idx_map = {}
        for i, v in enumerate(nums):
            complement = target - v
            if complement in idx_map:
                return [idx_map[complement], i]
            idx_map[v] = i

File: cpp/1-two-sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        map<int, int> store;

        for (int i = 0; i < nums.size(); i++)
        {
            int comp = target - nums[i];
            if (store.count(comp))
                return {store[comp], i};
            else
                store[nums[i]] = i;
        }

        return {};
    }
};